File: leetcode/etc/k-closest-points-to-origin.cpp
class Solution {
public:
    static bool compare(const vector<int>& a, const vector<int>& b) {
        double aDist = a[0] * a[0] + a[1] * a[1];
        double bDist = b[0] * b[0] + b[1] * b[1];
        return (aDist < bDist);
    }
    vector<vector<int>> kClosest(vector<vector<int>>& points, int K) {
        sort(points.begin(), points.end(), compare); 
        points.resize(K);
        return points;
    }
};

File: leetcode/mock-interview/k-closest-points-to-origin.js
/**
 * @param {number[][]} points
 * @param {number} K
 * @return {number[][]}
 */
const compare = function(a, b) {
    aDist = a[0] * a[0] + a[1] * a[1];
    bDist = b[0] * b[0] + b[1] * b[1];
    return aDist - bDist;
}
const kClosest = function(points, K) {
    return points.sort(compare).slice(0, K);
}
